mat64/dense.go

```diff
@@ -253,9 +254,10 @@ func (m *Dense) View(i, j, r, c int) Matrix {
 	return &t
 }
 
-// Grow returns an expanded copy of the receiver. The copy is expanded
-// by r rows and c columns. If the dimensions of the new copy are outside
-// the caps of the receiver a new allocation is made, otherwise not.
+// Grow returns the receiver expanded by r rows and c columns. If the dimensions
+// of the expanded matrix are outside the capacities of the receiver a new
+// allocation is made, otherwise not. Note the receiver itself is not modified
+// during the call to Grow.
```

Comment:
At some point when we start writing wiki-type docs this probably needs expanding. The whole "slice is a reference type" thing will rear its head here too - people wondering why the receiver sometimes reflects changes in the returned matrix and sometimes not.

Probably what we need to do is write a gonum-tour set of articles (I think I might have mentioned this previously).